libs/state/docs/snippets/global-state-vs-local-state.md

```diff
@@ -31,18 +31,172 @@ Let's take a look at a simple todo app. This app has 2 views.
+    }
+}
+```
+
 ### Setup
 
 - Renders a list of all existing `tasks` and a `counter` that shows the total amount of tasks.
 - The list can be expanded or collapsed and has property `isExpanded`.
 - Gets tasks as array from endpoint _tasks/get_.
 
+```typescript
+interface AllTodosState {
+    tasks: Task[];
+    isExpanded: boolean;
+}
+
+@Component({
+    selector: "all-tasks",
+    templateUrl: "./all-tasks.component.html"
+})
+export class AllTasksComponent extends RxState<AllTodosState>{
+    readonly tasks$ = this.select("tasks");
+    readonly counter$ = this.select(pluck("tasks"), map(tasks => tasks.length));
+    readonly isExpanded$ = this.select("isExpanded");
+
+    constructor(private tasksService: TasksService) {
+        super();
+
+        /* Fetch tasks from backend */
+        this.connect("tasks", this.tasksService.fetchTasks());
+    }
+}
+```
+
 ### What is global and what is local?
+Looking at the above examples, let us see what is **local** and what is **global**! 
 
 - `counter` property is a part of **local** state of each view. The counter value is specific for each view.
 - `isExpanded` property is also part of **local** state. Both lists can be expanded/collapsed but this status isn't shared between them and they don't care about this status of each other.
 - `tasks` array is a part of our app **global** state. This array needed for each view and received from the same endpoint. We don't need to load it twice. It is time to introduce a global layer to our application and move tasks array and retrieving logic there.
 
+### Moving the `tasks` array to our **global** state
+We can handle **global** state in different ways, but for this snippet we´re going to use an `injectionToken`.
+
+```typescript
+import { InjectionToken } from "@angular/core";
+import { RxState } from "@rx-angular/state";
+
+export interface Task {
+    id: number;
+    label: string;
+    done: boolean;
+}
+
+export interface GlobalState {
+    tasks: Task[];
+}
+
+export const GLOBAL_RX_STATE = new InjectionToken<RxState<GlobalState>>('GLOBAL_RX_STATE');
+```
+
+We then _provide_ the `injectionToken` in our `app.module.ts`.
+
+```typescript
+import { GLOBAL_RX_STATE, GlobalState } from "./rx-state";
+...
+
+@NgModule({
+    imports: [...],
+    declarations: [...],
+    providers: [{
+        provide: GLOBAL_RX_STATE, useFactory: () => new RxState<GlobalState>()
+    }],
+    bootstrap: [...]
+})
+export class AppModule {}
+```
+
+We can then provide the initial `tasks` in the `AppComponent` and just have our `TodoComponent` and `AllTasksComponent` connect to the global state.
+
+```typescript
+import { GLOBAL_RX_STATE } from './rx-state';
+
+@Component({
+  selector: 'my-app',
+  templateUrl: './app.component.html',
+  styleUrls: [ './app.component.css' ]
+})
+export class AppComponent  {
+constructor(@Inject(GLOBAL_RX_STATE) private state, private tasksService: TasksService) {
```

Comment:
How are those 2 services related to ech other? Also the state is not used..

Reply:
Inside the `constructor` there is the `connect` to the tasksService.

```
    /* Fetch tasks from backend */
    this.connect("tasks", this.tasksService.fetchTasks());
```

I think my original thought was that if two components are using the same state, they shouldn´t both call the API, so the parent should. At least from my memory.

If anything needs changing I'll do just that.